src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.ts

```diff
@@ -96,6 +103,63 @@ export class FileUploadExerciseUpdateComponent implements OnInit {
                 this.goBackAfterSaving = true;
             }
         });
+        this.activatedRoute.url
+            .pipe(
+                tap(
+                    (segments) =>
+                        (this.isImport = segments.some((segment) => segment.path === 'import', (this.isExamMode = segments.some((segment) => segment.path === 'exercise-groups')))),
+                ),
+                switchMap(() => this.activatedRoute.params),
+                tap((params) => {
+                    if (!this.isExamMode) {
+                        this.exerciseCategories = this.fileUploadExercise.categories || [];
+                        if (this.fileUploadExercise.course) {
+                            this.courseService.findAllCategoriesOfCourse(this.fileUploadExercise.course!.id!).subscribe({
+                                next: (categoryRes: HttpResponse<string[]>) => {
+                                    this.existingCategories = this.exerciseService.convertExerciseCategoriesAsStringFromServer(categoryRes.body!);
+                                },
+                                error: (error: HttpErrorResponse) => onError(this.alertService, error),
+                            });
+                        } else {
+                            this.courseService.findAllCategoriesOfCourse(this.fileUploadExercise.exerciseGroup!.exam!.course!.id!).subscribe({
+                                next: (categoryRes: HttpResponse<string[]>) => {
+                                    this.existingCategories = this.exerciseService.convertExerciseCategoriesAsStringFromServer(categoryRes.body!);
+                                },
+                                error: (error: HttpErrorResponse) => onError(this.alertService, error),
+                            });
+                        }
+                    } else {
+                        // Lock individual mode for exam exercises
+                        this.fileUploadExercise.mode = ExerciseMode.INDIVIDUAL;
+                        this.fileUploadExercise.teamAssignmentConfig = undefined;
+                        this.fileUploadExercise.teamMode = false;
+                        // Exam exercises cannot be not included into the total score
+                        if (this.fileUploadExercise.includedInOverallScore === IncludedInOverallScore.NOT_INCLUDED) {
+                            this.fileUploadExercise.includedInOverallScore = IncludedInOverallScore.INCLUDED_COMPLETELY;
+                        }
```

Comment:
This should probably be moved for the server, also for the other exercise types.